tests/common/snappi/snappi_helpers.py

```diff
@@ -238,3 +239,73 @@ def get_dut_port_id(dut_hostname, dut_port, conn_data, fanout_data):
            return i
 
     return None
+
+
+def wait_for_arp(snappi_api, max_attempts=10, poll_interval_sec=1):
+    """
+    This function waits for arp to get resolved for ipv4/ipv6 in the snappi config
+
+    Args:
+    snappi_api: snappi api
+    max_attempts: maximum attempts for timeout
+    poll_interval_sec: interval poll second
+
+    Return:
+    returns number of attempts if arp is resolved within max attempts else fail 
+    """
+    attempts = 0
+    v4_gateway_macs_resolved = False
+    v6_gateway_macs_resolved = False
+
+    get_config = snappi_api.get_config()
+    v4_addresses = []
+    v6_addresses = []
+
+    for device in get_config.devices:
+        for ethernet in device.ethernets:
+            for v4_address in ethernet.ipv4_addresses:
+                v4_addresses.append(v4_address.address)
+            for v6_address in ethernet.ipv6_addresses:
+                v6_addresses.append(v6_address.address)
+
+    while attempts < max_attempts:
+        request = snappi_api.states_request()
+        request.choice = request.IPV4_NEIGHBORS
+        states = snappi_api.get_states(request)
+
+        if len(v4_addresses) > 0:
+            v4_link_layer_address = [
+                state.link_layer_address
+                for state in states.ipv4_neighbors
+                if state.link_layer_address
```

Comment:
link_layer_address == True or link_layer_address is not None?

Reply:
link_layer_address will be mac address, if mac is resolved the get_states will return mac address or returns None

Reply:
> link_layer_address will be mac address, if mac is resolved the get_states will return mac address or returns None

Can you change the code to "if state.link_layer_address is not None"？

Reply:
ok..updated